### PR TITLE
cksum: read the next file when the first is missing or invalid

### PR DESCRIPTION
**Pull request by sylvestre**
Found by a wip fuzzer

### CODE REVIEW

File: tests/by-util/test_cksum.rs [hunk 1]
```diff
@@ -1221,3 +1221,32 @@ fn test_check_base64_hashes() {
         .succeeds()
         .stdout_is("empty: OK\nempty: OK\nempty: OK\n");
 }
+
+#[test]
+fn test_several_file_error_mgmt() {
```
Comment (cakebaker) [inline tests/by-util/test_cksum.rs:1226]:
```suggestion
fn test_several_files_error_mgmt() {
```

File: src/uucore/src/lib/features/checksum.rs [hunk 2]
```diff
@@ -462,6 +462,8 @@ pub fn perform_checksum_validation<'a, I>(
 where
     I: Iterator<Item = &'a OsStr>,
 {
+    let mut encountered_errors = false;
```
Comment (cakebaker) [inline src/uucore/src/lib/features/checksum.rs:465]: I think this variable is not necessary and you could use `set_exit_code(1)` instead.

Comment (sylvestre) [reply src/uucore/src/lib/features/checksum.rs]: smart, thanks!

**Comment by sylvestre** [comment]
as it wasn't covered by upstream, i added these tests:
https://github.com/coreutils/coreutils/commit/d3b4bc30803becfc383b39c25093d5be16f0a635

**Pull request merged**